machine-learning/app/config.py

```diff
@@ -0,0 +1,25 @@
+from pydantic import BaseSettings
+from functools import lru_cache
+
+class Settings(BaseSettings):
+    cache_folder: str = "/cache"
+    classification_model: str = "microsoft/resnet-50"
+    clip_image_model: str = "clip-ViT-B-32"
+    clip_text_model: str = "clip-ViT-B-32"
+    facial_recognition_model: str = "buffalo_l"
+    min_tag_score: float = 0.9
+    eager_startup: bool = True
+    model_ttl: int = 300
+    host: str = "0.0.0.0"
+    port:int = 3003
+    workers:int = 1
+    min_face_score: float = 0.7
+
+    class Config(BaseSettings.Config):
+        env_prefix = 'MACHINE_LEARNING_'
+        case_sensitive = False
+
+
+@lru_cache()
```

Comment:
I was trying to reduce code execution on imports, though I know here the effect is negligible, but still it forces us to write code in a better(IMO) way.

All the endpoints get this via the dependency management of FastAPI, other places we can get the object by calling this function, which is cached by the decorator `lru_cache`, so in effect we will have one object per process.

Reply:
I see where you're come from, but `lru_cache` is really overkill for this. When I see it I assume there's a reason for it to be there.

Plus `Settings` is still a `BaseModel`, so it's useful to change it after it's created and have it apply to all places where it's used. This is relevant since there are plans to make these configs dynamically changeable through an endpoint.

Reply:
yeah It doesn't do anything funky, just reuses the object. I don't know why you don't like him :)

so the idea of having a function is fine, but not caching it. right?

Reply:
Eh, to me it's a code smell. It's a function that takes no arguments and just calls `Settings()` with no arguments. It caches but there's no compute-heavy logic either. At that point it doesn't really have a reason to exist.

Reply:
okay will get rid of it, but I still feel import time processing should be avoided!